### PR TITLE
docs: add blog post creating an extension

### PR DESCRIPTION
**Pull request by cdrage**
docs: add blog post creating an extension

### What does this PR do?

- Adds a blog post on how to create an extension
- Generic and points to the documentation we already have, but presented
  in a blog-style way

### Screenshot / video of UI

N/A, see netlify

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop-internal/issues/354

### How to test this PR?

N/A

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
**DO NOT MERGE YET**

cc @Firewall @slemeur (see latest comment: https://github.com/containers/podman-desktop-internal/issues/354#issuecomment-2430225157)

File: website/blog/2024-10-21-creating-an-extension.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+title: How to create your first extension
+description: Learn how to create and customize your own extension for Podman Desktop
+slug: extensions-getting-started
+authors: [cdrage]
+tags: [podman-desktop, release, podman]
```
Comment (Firewall) [inline website/blog/2024-10-21-creating-an-extension.md]:
```suggestion
tags: [podman-desktop, release, podman, extension, plugin]
```

**Comment by axel7083** [comment]
> DO NOT MERGE YET

keep it as a draft ?

File: website/blog/2024-10-21-creating-an-extension.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+In this guide, we'll introduce how you can build your own Podman Desktop extension, with links to detailed documentation that covers each part of the process.
+
+## Introduction to Extensions
+
+Extensions are abundant in Podman Desktop and can be found in the **Extensions -> Catalog** section.
+
+![extension catalog](img/podman-desktop-release-1.10/extension-catalog.png)
+
+Each extension expands on Podman Desktop, such as providing [Kubernetes development clusters with Minikube](https://github.com/containers/podman-desktop-extension-minikube) or even [analyzing your images](https://github.com/containers/podman-desktop-extension-layers-explorer).
```
Comment (axel7083) [inline website/blog/2024-10-21-creating-an-extension.md]:
```suggestion
Each extension expands on Podman Desktop, such as providing [Kubernetes development clusters with Minikube](https://github.com/containers/podman-desktop-extension-minikube) or even [analyzing your images layers](https://github.com/containers/podman-desktop-extension-layers-explorer).
```

With `analyzing your images` I though about security, when hovering the links I understood you were talking about the layers explorer extension

File: website/blog/2024-10-21-creating-an-extension.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+Below we will go over how to start creating your first extension!
+
+## Getting Started with Your Project
+
+The first step in creating your extension is setting up the project environment. To learn how to configure the project and add basic components, check out the **[Templates for creating an extension](/docs/extensions/templates)** guide, which walks you through initializing your project from an official template.
+
+## Adding UI Components
```
Comment (axel7083) [inline website/blog/2024-10-21-creating-an-extension.md]: I feel like we are missing a section here something like `architecture` to give some insight of how an extension works, and maybe the difference with the dockers extension which run in containers, where podman extensions are running with node on the host machine.

Comment (axel7083) [reply website/blog/2024-10-21-creating-an-extension.md]: Maybe adding something about the possibility of creating an extension without UI?

File: website/blog/2024-10-21-creating-an-extension.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+One of the most common tasks when creating an extension is enhancing the user interface. Whether it’s adding buttons, panels, or icons, UI components help make your extension more interactive and accessible. Learn more about this in the **[Adding UI components](/docs/extensions/developing/adding-ui-components)** documentation, where you’ll find instructions on creating and integrating components into the application’s UI.
+
+### Working with Icons
+
+Icons are a great way to make your extension more visually unique. You can learn how to add and style custom icons by following the **[Adding icons](/docs/extensions/developing/adding-icons)** documentation.
```
Comment (axel7083) [inline website/blog/2024-10-21-creating-an-extension.md]: I would add a screenshot of the kind icons to illustrate this concept, what do you think?

File: website/blog/2024-10-21-creating-an-extension.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+### Menus and Navigation
+
+Extensions often integrate with existing menus and navigation to offer users easy access to new commands and features. If you want to add items to the context menu, explore the **[Menu configuration](/docs/extensions/developing/menu)** documentation, which explains how to add commands to menus and control when they are displayed using When Clauses.
+
+## Adding and Configuring Commands
+
+Commands are the backbone of most extensions, allowing users to interact with the application and trigger specific actions.
```
Comment (axel7083) [inline website/blog/2024-10-21-creating-an-extension.md]: Commands are heavily inspired from vscode, might be a good comparison, for developer which are familiar with vscode extensions

Comment (slemeur) [reply website/blog/2024-10-21-creating-an-extension.md]: +1. 
I'd also like to callout a specific example. I think that can nicely illustrate the paragraph

File: website/blog/2024-10-21-creating-an-extension.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+One of the most common tasks when creating an extension is enhancing the user interface. Whether it’s adding buttons, panels, or icons, UI components help make your extension more interactive and accessible. Learn more about this in the **[Adding UI components](/docs/extensions/developing/adding-ui-components)** documentation, where you’ll find instructions on creating and integrating components into the application’s UI.
+
+### Working with Icons
+
+Icons are a great way to make your extension more visually unique. You can learn how to add and style custom icons by following the **[Adding icons](/docs/extensions/developing/adding-icons)** documentation.
+
+### Menus and Navigation
+
+Extensions often integrate with existing menus and navigation to offer users easy access to new commands and features. If you want to add items to the context menu, explore the **[Menu configuration](/docs/extensions/developing/menu)** documentation, which explains how to add commands to menus and control when they are displayed using When Clauses.
```
Comment (slemeur) [inline website/blog/2024-10-21-creating-an-extension.md]: Should we show an example, like the menu of AI Lab?

File: website/blog/2024-10-21-creating-an-extension.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Commands are the backbone of most extensions, allowing users to interact with the application and trigger specific actions. 
+
+If you need to define and register custom commands, the **[Commands](/docs/extensions/developing/commands)** guide will show you how to create commands that respond to user actions or input, and tie them into your extension’s workflow.
+
+You can also configure these commands to appear in different contexts. Check out the **[When clause Contexts](/docs/extensions/developing/when-clause-context)** documentation to learn more about restricting commands to specific scenarios.
+
+## Setting Up Onboarding Workflows
+
+Creating a smooth onboarding experience is essential to help users get started with your extension. This includes steps for CLI binary installations or other initial setup values.
```
Comment (slemeur) [inline website/blog/2024-10-21-creating-an-extension.md]: A screenshot here again could be beneficial for the user to understand what can be done

**Review by slemeur — CHANGES_REQUESTED** [review]

File: website/blog/2024-10-21-creating-an-extension.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
+---
+title: How to create your first extension
```
Comment (slemeur) [inline website/blog/2024-10-21-creating-an-extension.md]: Introduction to Podman Desktop extensions

File: website/blog/2024-10-21-creating-an-extension.md (hunk 3 above)
Comment (cdrage) [reply website/blog/2024-10-21-creating-an-extension.md]: Created https://github.com/containers/podman-desktop/issues/9593

File: website/blog/2024-10-21-creating-an-extension.md (hunk 6 above)
Comment (cdrage) [reply website/blog/2024-10-21-creating-an-extension.md]: Menu in AI lab is a bit different, on the document linked it explains how this is more for "show actions" sections of an image (how we do it with bootc).

File: website/blog/2024-10-21-creating-an-extension.md (hunk 4 above)
Comment (cdrage) [reply website/blog/2024-10-21-creating-an-extension.md]: Added!

File: website/blog/2024-10-21-creating-an-extension.md (hunk 6 above)
Comment (cdrage) [reply website/blog/2024-10-21-creating-an-extension.md]: Added an image!

File: website/blog/2024-10-21-creating-an-extension.md (hunk 5 above)
Comment (cdrage) [reply website/blog/2024-10-21-creating-an-extension.md]: Gave links to the documentation as well as talked about VS Code :)

File: website/blog/2024-10-21-creating-an-extension.md (hunk 7 above)
Comment (cdrage) [reply website/blog/2024-10-21-creating-an-extension.md]: Added screenshot from our compose onboarding

**Comment by cdrage** [comment]
Thanks for the feedback @Firewall @slemeur 

I have added more screenshots, added changes based on every suggestions in a new commit and the PR is ready for review again.

**Review by slemeur — APPROVED** [review]

**Comment by cdrage** [comment]
@Firewall @slemeur 

What date works for publishing the blog / suggestions?

**Review by Firewall — APPROVED** [review]

**Comment by cdrage** [comment]
Renamed the file to reflect today's date and setting this as auto-merge (as per feedback from standup call)

**Pull request merged**